src/evm.rs

```diff
@@ -43,11 +43,20 @@ pub struct CallRawResult {
 
 impl From<CallTraceNode> for CallTrace {
     fn from(item: CallTraceNode) -> Self {
+        let data = match item.trace.kind {
+            CallKind::Call | CallKind::StaticCall => {
+                let first_4_bytes: Vec<u8> =
```

Comment:
In Rust, it's often more idiomatic to use slices (&[u8]) or arrays ([u8; N]) instead of vectors (Vec<u8>) when working with fixed-size chunks of data. In this specific case, you are taking the first 4 bytes of raw data, and the size is fixed at 4 bytes.

```rust
let first_4bytes: [u8; 4] = item.trace.data.to_raw().iter().take(4).cloned().collect::<Vec<u8>>().try_into().unwrap();
```